### PR TITLE
Update more commands to use external versions

### PR DESCRIPTION
**Pull request by juanvallejo**
~~Work in progress~~

cc @soltysh

### CODE REVIEW

File: pkg/oc/cli/experimental/config/patch.go [hunk 1]
```diff
@@ -50,8 +50,15 @@ var (
 		%[1]s openshift.local.config/master/master-config.yaml --patch='{"auditConfig": {"enabled": true}}'`)
 )
 
+func NewPatchOptions(streams genericclioptions.IOStreams) *PatchOptions {
+	return &PatchOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("patched").WithTypeSetter(configapi.Scheme).WithDefaultOutput("yaml"),
```
Comment (juanvallejo) [inline pkg/oc/cli/experimental/config/patch.go]: @deads2k I'm unfamiliar with this scheme, but it is the one this command uses with the resource builder so I am using it here as well

Comment (soltysh) [reply pkg/oc/cli/experimental/config/patch.go]: Yes, this scheme has only the config types, which is the only types needed for this command to know.

File: pkg/oc/cli/process/process.go [hunk 2]
```diff
@@ -25,18 +25,23 @@ import (
...
 	"github.com/openshift/origin/pkg/template/apis/template"
+	internaltemplateapi "github.com/openshift/origin/pkg/template/apis/template"
+	templateapiv1 "github.com/openshift/origin/pkg/template/apis/template/v1"
 	templatevalidation "github.com/openshift/origin/pkg/template/apis/template/validation"
-	templateinternalclient "github.com/openshift/origin/pkg/template/client/internalversion"
-	templateclientinternal "github.com/openshift/origin/pkg/template/generated/internalclientset"
-	templateclient "github.com/openshift/origin/pkg/template/generated/internalclientset/typed/template/internalversion"
+	//templateinternalclient "github.com/openshift/origin/pkg/template/client/internalversion"
+	//templateclientinternal "github.com/openshift/origin/pkg/template/generated/internalclientset"
+	//templateclient "github.com/openshift/origin/pkg/template/generated/internalclientset/typed/template/internalversion"
```
Comment (soltysh) [inline pkg/oc/cli/process/process.go]: Remove these.

File: pkg/oc/cli/process/process.go [hunk 3]
```diff
@@ -25,18 +25,23 @@ import (
...
 
 	templateapi "github.com/openshift/api/template"
-	templateapiv1 "github.com/openshift/api/template/v1"
+	templatev1 "github.com/openshift/api/template/v1"
 	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 	"github.com/openshift/origin/pkg/oc/lib/describe"
 	"github.com/openshift/origin/pkg/oc/lib/newapp/app"
 	"github.com/openshift/origin/pkg/oc/util/ocscheme"
 	"github.com/openshift/origin/pkg/template/apis/template"
+	internaltemplateapi "github.com/openshift/origin/pkg/template/apis/template"
```
Comment (soltysh) [inline pkg/oc/cli/process/process.go]: `templateapi` is ok for these.

Comment (soltysh) [reply pkg/oc/cli/process/process.go]: It goes logically with the latter `templateapiv1` you have.

File: pkg/oc/cli/process/process.go [hunk 4]
```diff
@@ -253,16 +258,16 @@ func (o *ProcessOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args [
 		}
 
 		// TODO: switch to using external templates client
-		templateClientset, err := templateclientinternal.NewForConfig(clientConfig)
+		templateClientset, err := templatev1client.NewForConfig(clientConfig)
 		if err != nil {
 			return err
 		}
-		o.templateClient = templateClientset.Template()
+		o.templateClient = templateClientset
```
Comment (soltysh) [inline pkg/oc/cli/process/process.go]: Assign to `o.templateClient`, don't use that temporary variable.

File: pkg/oc/cli/process/process.go [hunk 5]
```diff
@@ -479,17 +489,32 @@ func injectUserVars(values app.Environment, t *template.Template, ignoreUnknownP
...
-	if errs := templatevalidation.ValidateProcessedTemplate(tpl); len(errs) > 0 {
+func processTemplateLocally(tpl *templatev1.Template) (*templatev1.Template, error) {
+	// TODO: Create validation helpers unique to the client
+	// We shouldn't be using api helpers anyway
+	internalTemplate := &internaltemplateapi.Template{}
+	if err := templateapiv1.Convert_v1_Template_To_template_Template(tpl, internalTemplate, nil); err != nil {
+		return nil, err
+	}
+
+	if errs := templatevalidation.ValidateProcessedTemplate(internalTemplate); len(errs) > 0 {
```
Comment (soltysh) [inline pkg/oc/cli/process/process.go]: That sucks :/

**Comment by juanvallejo** [comment]
Other than that, nothing else changed and everything else is good to go on this PR

**Pull request merged**